src/dev/amdgpu/sdma_engine.cc

```diff
@@ -653,11 +654,27 @@ SDMAEngine::writeDone(SDMAQueue *q, sdmaWrite *pkt, uint32_t *dmaBuffer)
 {
     DPRINTF(SDMAEngine, "Write packet completed to %p, %d dwords\n",
             pkt->dest, pkt->count);
-    delete []dmaBuffer;
+
+    auto cleanup_cb = new EventFunctionWrapper(
+        [ = ]{ writeCleanup(dmaBuffer); }, name());
+
+    auto system_ptr = gpuDevice->CP()->system();
+    if (!system_ptr->isAtomicMode()) {
+        warn_once("SDMA cleanup assumes 2000 tick timing for completion."
+                " This has not been testing in timing mode\n");
+    }
+    schedule(cleanup_cb, curTick() + 10000);
```

Comment:
above print says 2K ticks, but seems like 10K here?

Reply:
Added a comment here. 2K is the minimum required wait time, but I've added more in case this changes (e.g., if you change the GPU frequency it could)